### DISCUSSION TITLE
Multi gNBs to a single UE in ZMQ

### DISCUSSION THREAD

**Discussion by hududed** [srsran/srsRAN_Project#905] [General Help] [answered]
I was wondering if its possible to setup multiple gNBs to connect to a single UE. I was following [this](https://github.com/srsran/srsRAN_Project/discussions/561) not using docker but everything runs on localhost. I changed the `amf.bind_addr` for gNB2 to a different address (127.0.0.5 and 127.0.0.8), the TX and RX ports are also different. 

```
10/29 11:19:35.965: [amf] INFO: gNB-N2 accepted[127.0.0.5]:52854] in ng-path module (../src/amf/ngap-sctp.c:113) 
10/29 11:19:35.965: [amf] INFO: gNB-N2 accepted[127.0.0.5] in master_sm module (../src/amf/amf-sm.c:796) 
10/29 11:19:35.972: [amf] INFO: [Added] Number of gNBs is now 1 (../src/amf/context.c:1238) 
10/29 11:19:35.972: [amf] INFO: gNB-N2[127.0.0.5] max_num_of_ostreams : 30 (../src/amf/amf-sm.c:842) 
10/29 11:19:38.238: [amf] INFO: gNB-N2 accepted[127.0.0.8]:35359] in ng-path module (../src/amf/ngap-sctp.c:113) 
10/29 11:19:38.238: [amf] INFO: gNB-N2 accepted[127.0.0.8] in master_sm module (../src/amf/amf-sm.c:796) 
10/29 11:19:38.251: [amf] INFO: [Added] Number of gNBs is now 2 (../src/amf/context.c:1238) 
10/29 11:19:38.251: [amf] INFO: gNB-N2[127.0.0.8] max_num_of_ostreams : 30 (../src/amf/amf-sm.c:842) 
```
The gNBs connect to the core, but the UE is stuck at `Attaching UE...`

Do I have to add the bridge as below? If so, how do I do this on localhost
```
Get the name of the Open5gs docker bridge:
ip -o addr show | grep "10.53.1.1"
Add a second IP address to the bridge:
sudo ip addr add 10.53.1.3/24 dev BRIDGE_NAME
```

I have also attached [ue_.log](https://github.com/user-attachments/files/17561929/ue_.log)

**Comment by pgawlowicz** [comment, marked as answer]
Did you connect 2 gnbs and a UE over a gnuradio signal broker? You cannot directly connect one UE to multiple GNBs. 
Also, since both have to be on the same EARFCN, you will probably have to apply a much bigger path loss to one of them. 
Similar as done [here](https://docs.srsran.com/projects/4g/en/latest/app_notes/source/handover/source/index.html#s1-handover) for the 4G handover.
Note that srsUE does not support 5G HO.

**Reply by hududed** [reply]
Yes using gnuradio ZMQ as broker. Ok, so for clarity - it is possible to connect 1 UE to multiple eNBs (both using srsRAN_4G), if one has significantly higher path loss than the other. But this is not possible for 1 UE (srsUE from srsRAN_4G) to multiple gNBs (srsRAN_Project), correct?

Is there a reason for this? And are there plans to change this in the near future?

**Reply by pgawlowicz** [reply]
Once the 5G srsUE connects to one gnb, it will not switch to the other one, as the 5G handover is not implemented.
